event_feed.go

```diff
@@ -16,17 +17,29 @@ type EventFeed struct {
 	lastCursor string
 }
 
-func newEventFeed(client *Client, source EventSource, opts ...FeedOptFn) (*EventFeed, error) {
+func newEventFeed(client *Client, source EventSource, fromQuery bool, opts ...FeedOptFn) (*EventFeed, error) {
 	feed := &EventFeed{
 		client: client,
 		source: source,
 		opts:   opts,
 	}
 
+	// init a feed request to validate feed options
+	req, err := feed.newFeedRequest()
+	if err != nil {
+		return nil, err
+	}
+	if fromQuery && len(req.Cursor) > 0 {
+		return nil, fmt.Errorf("cannot use EventFeedCursor with FeedFromQuery")
```

Comment:
This validation breaks the boundary here. It's FeedFromQuery that cares about this, not newEventFeed. Let's keep that logic in FeedFromQuery and just make sure they aren't both set here.